archive/rule.go

```diff
@@ -1,34 +1,53 @@
 package archive
 
 import (
-	"errors"
 	"fmt"
 	"path/filepath"
 	"strings"
 
 	"github.com/brimsec/zq/ast"
 	"github.com/brimsec/zq/zng/resolver"
+	"github.com/brimsec/zq/zqe"
 	"github.com/brimsec/zq/zql"
 )
 
-func ParsePattern(in string) (string, string, error) {
+type IndexQuery struct {
+	indexName string
+	patterns  []string
+}
+
+func ParseIndexQuery(indexName string, patterns []string) (iq IndexQuery, err error) {
+	if len(patterns) == 0 {
+		return iq, zqe.E(zqe.Invalid, "no search patterns")
+	}
+	if indexName != "" {
+		iq.indexName = indexName
+		iq.patterns = patterns
+		return
```

Comment:
Nit: There's no reason to use a naked return here or name the result parameters, so don't.